gateway/api_definition.go

```diff
@@ -1313,3 +1318,18 @@ func (r *RoundRobin) WithLen(len int) int {
 	cur := atomic.AddUint32(&r.pos, 1) - 1
 	return int(cur) % len
 }
+
+var listenPathVarsRE = regexp.MustCompile(`{[^:]+(:[^}]+)?}`)
+
+func stripListenPath(listenPath, path string, muxVars map[string]string) string {
+	if !strings.Contains(listenPath, "{") {
+		return strings.TrimPrefix(path, listenPath)
+	}
+	lp := listenPathVarsRE.ReplaceAllStringFunc(listenPath, func(match string) string {
+		match = strings.TrimLeft(match, "{")
+		match = strings.TrimRight(match, "}")
```

Comment:
We do not really care about what is on the right side of ":", so we can safely skip this line.

Reply:
Additionally, we can just replace this both lines by: `aliasVar := strings.Split(match[1:], ":")[0]`
We know that first character is always `{`

Reply:
Just add code comment clarifying this code.

Reply:
This takes into account for the case of `{key}` If we don't do that then `aliasVar="key}"`

Reply:
I see, no worries, thats fine then